src/service/git/github-api.ts

```diff
@@ -93,21 +97,26 @@ export class GithubAPIService {
     try {
       // ensure that repo exists
       if (targetOwner === sourceOwner) {
+        this.logger.debug(`Making a github API call to check whether ${sourceOwner}/${repo} exists`);
         await this.octokit.repos.get({
           owner: sourceOwner,
           repo,
         });
         return repo;
       } else {
         // find repo from fork list
+        let page = 1;
         for await (const response of this.octokit.paginate.iterator(this.octokit.repos.listForks, {
           owner: targetOwner,
           repo,
+          per_page: 100
```

Comment:
```suggestion
          per_page: 100,
          page          
```

Reply:
page is missing, right?

Reply:
I would even extract this find repo from fork part block to different private method

Reply:
The iterator handles the `page` part. I am using the variable `page` for logging purposes just so that we know till what page we made the requests to